.github/workflows/crucible-go-build.yml

```diff
@@ -14,15 +14,15 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-24.04]
-        cabal: ["3.10.3.0"]
-        ghc: ["9.4.8", "9.6.5", "9.8.2"]
+        cabal: ["3.14.2.0"]
+        ghc: ["9.6.5", "9.8.2", "9.10.1"]
```

Comment:
Any reason not to use the latest minor releases (9.6.7 and 9.8.4) here?

Reply:
Ugh, this has turned out to be a bit more of a pain than I thought, because the most recent release of NixOS doesn't have these compilers:
```
error: flake 'github:nixos/nixpkgs/nixos-24.11' does not provide attribute 'packages.x86_64-linux.haskell.compiler.ghc967', 'legacyPackages.x86_64-linux.haskell.compiler.ghc967' or 'haskell.compiler.ghc967'
       Did you mean one of ghc927, ghc947, ghc96, ghc963 or ghc964?
```
Perhaps I'll do #1368 first and rebase...